### PR TITLE
Fix the connectors and ports, when the group is collapsed.

### PR DESCRIPTION
**Pull request by reddyashish**
### Purpose

This PR is to fix the issue related to connectors and ports when the group is collapsed. 

Task: https://jira.autodesk.com/browse/DYN-3983

group collapsing

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhacement. **Mandatory section** 

### Reviewers

@QilongTang @SHKnudsen

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs [hunk 1]
```diff
@@ -649,6 +649,39 @@ internal IEnumerable<PortModel> GetGroupOutPorts(IEnumerable<NodeModel> ownerNod
...
+            int index = portModel.LineIndex == -1 ? portModel.Index : portModel.LineIndex;
+
+            //If the port was not found, then it should have just been deleted. Return from function
+            if (index == -1)
+                return verticalOffset;
+
+            // calculate the vertical offset for the port index.
+            int portVerticalMidPoint = 17;
+            double portHeight = portModel.Height;
+            return verticalOffset + (index * portHeight) + portVerticalMidPoint;
```
Comment (SHKnudsen) [inline src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs:665]: Will this work if there are Ports from multiple nodes on the group?

The index is relative to the owner node and not the group, right?

Comment (reddyashish) [reply src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs]: Yes, you are right. Modified it to use the port index on the group.

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs [hunk 2]
```diff
@@ -649,6 +649,39 @@ internal IEnumerable<PortModel> GetGroupOutPorts(IEnumerable<NodeModel> ownerNod
                 );
         }
 
+        private double GetPortVerticalOffset(PortModel portModel, int proxyPortIndex)
+        {
+            // vertical offset accounts for the port margins
+            double verticalOffset = 20;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs:655]: can you define the constants used as const in class?

File: src/DynamoCore/Graph/Nodes/PortModel.cs [hunk 3]
```diff
@@ -309,8 +323,11 @@ public bool IsConnected
                 return Connectors.Any() || (UsingDefaultValue && DefaultValue != null);
             }
         }
+
         #endregion
-        
+
+        internal bool IsProxyPort { get; set; } = false;
```
Comment (QilongTang) [inline src/DynamoCore/Graph/Nodes/PortModel.cs:329]: If we add this property here, should we remove the one in PortViewModel?

**Review by QilongTang — COMMENTED** [review]
Some comments

File: src/DynamoCore/Graph/Nodes/PortModel.cs (hunk 3 above)
Comment (reddyashish) [reply src/DynamoCore/Graph/Nodes/PortModel.cs:329]: yes, that is already removed.

Comment (QilongTang) [reply src/DynamoCore/Graph/Nodes/PortModel.cs:329]: Great, just saw it

**Comment by QilongTang** [comment]
Thank you @reddyashish , once the one remaining comment addressed, this LGTM

**Review by QilongTang — COMMENTED** [review]
LGTM, do you see any possibility to add unit tests to guard the position when opening a file with collapsed group?

**Comment by reddyashish** [comment]
Discussed with @QilongTang regarding the tests and decided that the test wouldn't be of much use in this case as the proxy ports position will be calculated internally based on the position of the annotation model. Merging this.

**Review by QilongTang — APPROVED** [review]

**Pull request merged**